### ISSUE TITLE
Reload source maps in the debugger when reloading the script environment

### ISSUE THREAD

**Issue by rroman6174** [Mojang/minecraft-debugger#130]
As a Minecraft script developer, I want the VSCode debugger to automatically reload source maps when the game is reloaded using the `/reload` command so that my breakpoints and other debugging tools continue to work correctly without requiring manual intervention.

#### Description
Currently, when I use the `/reload` command in Minecraft, the VSCode debugger does not reload the source maps. This causes breakpoints and other debugging tools to become misaligned or ineffective, which interrupts my workflow and requires me to manually reload the extension.

#### Acceptance Criteria
1. **Given** I have the Minecraft script debugger VSCode extension running and attached to a Minecraft game,
   **When** I execute the `/reload` command in Minecraft,
   **Then** the VSCode debugger should automatically reload the source maps.

2. **Given** I have set breakpoints in my script files before executing the `/reload` command,
   **When** the source maps are reloaded automatically,
   **Then** the breakpoints should remain accurate and functional without requiring manual adjustment or extension reload.

**Comment by MKokeshi** [comment]
To implement a reload command for a Minecraft script debugger in VSCode, you need to extend the functionality of the VSCode extension. Here’s a basic outline of the code snippets you might use to accomplish this, focusing on detecting the `/reload` command and reloading source maps.

### Step 1: Detect the `/reload` Command

You'll need to listen to the output of the Minecraft console to detect when the `/reload` command is executed. This can be done by extending the VSCode extension's functionality to listen for specific console output or events.

#### Example Code (Listening for `/reload` Command):

```javascript
const vscode = require('vscode');

function activate(context) {
    const debugSession = vscode.debug.activeDebugSession;

    if (debugSession) {
        const outputChannel = vscode.window.createOutputChannel('Minecraft Debugger');

        // Event listener for receiving output from the Minecraft console
        debugSession.addEventHandler('output', (output) => {
            // Check if the output contains the '/reload' command
            if (output.includes('/reload')) {
                reloadSourceMaps();
            }
        });
    }
}

function deactivate() {}

function reloadSourceMaps() {
    const debugSession = vscode.debug.activeDebugSession;

    if (debugSession) {
        debugSession.customRequest('reloadSourceMaps')
            .then(() => {
                vscode.window.showInformationMessage('Source maps reloaded successfully.');
            })
            .catch(err => {
                vscode.window.showErrorMessage(`Error reloading source maps: ${err.message}`);
            });
    }
}

module.exports = {
    activate,
    deactivate
};
```

### Step 2: Implement Reload Source Maps

In VSCode, there is no direct API for reloading source maps, so you may need to manually handle this by reloading or refreshing the relevant files or breakpoints.

#### Example Code (Refreshing Breakpoints):

```javascript
function reloadSourceMaps() {
    const debugSession = vscode.debug.activeDebugSession;

    if (debugSession) {
        // Refresh the breakpoints manually
        const breakpoints = debugSession.breakpoints;

        breakpoints.forEach(bp => {
            // Assuming you have a method to re-apply breakpoints
            debugSession.addBreakpoints(bp);
        });

        // Inform the user
        vscode.window.showInformationMessage('Source maps reloaded and breakpoints refreshed.');
    }
}
```

### Step 3: Update `launch.json` for Custom Request

Ensure that your `launch.json` configuration allows for custom requests. This might be necessary for the `reloadSourceMaps` function to work as intended.

#### Example `launch.json` Configuration:

```json
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Minecraft Debug",
            "type": "minecraft",
            "request": "launch",
            "program": "${workspaceFolder}/path/to/your/script",
            "preLaunchTask": "build",
            "customRequest": "reloadSourceMaps"
        }
    ]
}
```

### Step 4: Handling Debugger Requests

In your extension, you should define how the custom request for reloading source maps is handled. This typically involves setting up a debug adapter that responds to custom requests.

#### Example Code (Debug Adapter Handling):

```javascript
const vscode = require('vscode');

class MinecraftDebugAdapter {
    // ... other methods

    handleCustomRequest(request, response) {
        switch (request.command) {
            case 'reloadSourceMaps':
                this.reloadSourceMaps();
                response.success = true;
                break;
            default:
                response.success = false;
                response.message = `Unknown command: ${request.command}`;
        }
    }

    reloadSourceMaps() {
        // Logic to refresh source maps
    }
}

// Register the debug adapter
vscode.debug.registerDebugAdapterDescriptorFactory('minecraft', new MinecraftDebugAdapterFactory());
```

### Conclusion

These code snippets provide a starting point for implementing a reload command feature in a VSCode Minecraft debugging extension. The actual implementation may vary based on your specific debugging setup and the VSCode API capabilities. Ensure to adapt and extend these snippets as needed to fit the exact requirements of your extension and debugging environment.

**Comment by chmeyer-ms** [comment]
Fixed with #248
